File: DSA/Rahul_Mankani_590014725/Day29_Question3.c
#include <stdio.h>

int bit(int n, int i) 
{
    // Create mask with 1 at position i (1 << i)
    int mask = 1 << i;
    // Set i-th bit to 1 using OR and return result
    return n | mask;
}
int main() 
{
    int n, i;
    printf("Enter a number: ");
    scanf("%d", &n);
    printf("Enter bit position (0-indexed): ");
    scanf("%d", &i);
    int result = bit(n, i);
    printf("After setting bit %d, %d becomes: %d\n", i, n, result);
    return 0;
}

File: DSA/VaniJain_590012020/Day29Question3.java
import java.util.*;
public class SetBitAtPosition 
{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter an integer (n): ");
        int n = sc.nextInt();
        System.out.print("Enter the bit position to set (i): ");
        int i = sc.nextInt();
        int result = n | (1 << i);
        System.out.println("Number after setting " + i + "-th bit: " + result);
    }
}
